docs/how-to/contribute.rst

```diff
@@ -186,3 +233,37 @@ native widget capabilities to the Toga API. You may find it helpful to look at
 existing widgets to work out what is needed.
 
 Most importantly - have fun!
+
+Advanced Mode
+=============
+
+If you've got expertise in a particular platform (for example, if you've got
+experience writing iOS apps), or you'd *like* to have that experience, you
+might want to look into a more advanced problem. Here are some suggestions:
+
+* **Implement a platform native widget** If the core library already specifies
+  an interface, implement that interface; if no interface exists, propose an
+  interface design, and implement it for at least one platform.
+
+* **Add a new feature to an existing widget API** Can you think of a feature
+  than an existing widget should have? Propose a new API for that widget, and
+  provide a sample implementation.
+
+* **Improve platform specific testing** The tests that have been described in
+  this document are all platform independent. They use the dummy backend to
+  validate that data is being passed around correctly, but they don't validate
+  that on a given platform, widgets behave they way they should. If I put a
+  button on Toga app, is that button displayed? Is it in the right place? Does
+  it respond to mouse clicks? Ideally, we'd have automated tests to validate
+  these properties. However, automated tests of GUI operations can be difficult
+  to set up. If you've got experience with automated GUI testing, we'd love to
+  hear your suggestions.
+
+* **Improve the testing API for application writers** The dummy backend exists
```

Comment:
I really like this approach -- allowing authors to test their own apps using toga_dummy. As I understand it, you're asking contributors to make apps and write tests with Dummy  API to "feel" its limitations. If they can't figure out how to test some component of their app, they create an issue/PR with suggestions.

Reply:
That's the idea, yes. The dummy backend has very limited capabilities at present, driven mostly by our own internal testing needs. The longer-term aim is to make GUI apps as easy to test as any other app by providing the test tooling out of the box.